src/interface/container.cpp

```diff
@@ -234,7 +235,14 @@ VariableFluxPack<T> Container<T>::PackVariablesAndFluxesHelper_(
   auto key = std::make_pair(var_names, flx_names);
   auto kvpair = varFluxPackMap_.find(key);
   if (kvpair == varFluxPackMap_.end()) {
-    auto pack = MakeFluxPack(vars, fvars, &vmap);
+    // TODO(pgrete) please shout at me if this ugly piece is still in the code
```

Comment:
shouting

Reply:
Do people have other suggestions for this?
If this kind of hack is required to make the unit test runs, that hints at a design flaw.
So I'd prefer fixing the overall design rather than introducing a workaround.
Do people feel that it'd make sense to add an execution space to a `Variable`?

Reply:
I'm hesitant to add an exec space to a variable.  Since variables never live outside a container (at least in our current examples), it *may* make sense to add one to the container, but I'm not convinced of that either.  Additionally, typically all the variables we use are typically tied to a block and few are allocated such that they span blocks.  

All that is a long way of saying that I'm not sure it makes sense to add an exec space to every variable.

Reply:
What about moving this logic to `MakeFluxPack`? We could have it default to `DevExecSpace()` unless it's passed a value. We could even have it take a pointer to a meshblock and do the right thing if that pointer is not null.